packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectProject.tsx

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import "./SelectProject.scss";
+
+import { IModelApp } from "@bentley/imodeljs-frontend";
+import { AccessToken } from "@bentley/itwin-client";
+import { ProjectGrid } from "@itwin/imodel-browser-react";
+import {
+  SvgCalendar,
+  SvgList,
+  SvgSearch,
+  SvgStarHollow,
+} from "@itwin/itwinui-icons-react";
+import {
+  HorizontalTab,
+  HorizontalTabs,
+  IconButton,
+  LabeledInput,
+} from "@itwin/itwinui-react";
+import { useNavigate } from "@reach/router";
+import React, { useEffect, useState } from "react";
+
+import { SignIn } from "../signin/SignIn";
+
+const PROJECT_TYPE_MAP = ["", "?recents", "?myprojects"];
+
+const tabsWithIcons = [
+  <HorizontalTab
+    key="favorite"
+    label="Favorite projects"
+    startIcon={<SvgStarHollow />}
+  />,
+  <HorizontalTab
+    key="recents"
+    label="Recent projects"
+    startIcon={<SvgCalendar />}
+  />,
+  <HorizontalTab key="all" label="My projects" startIcon={<SvgList />} />,
+];
+
+export const SelectProject = () => {
+  const [projectType, setProjectType] = useState(() =>
+    PROJECT_TYPE_MAP.includes(window.location.search)
+      ? PROJECT_TYPE_MAP.indexOf(window.location.search)
+      : 0
+  );
+
+  const [searchValue, setSearchValue] = React.useState("");
+  const [searchParam, setSearchParam] = React.useState("");
+  const startSearch = React.useCallback(() => {
+    setSearchParam(searchValue);
+  }, [searchValue]);
+  const [accessToken, setAccessToken] = useState<AccessToken | undefined>();
```

Comment:
could just use the `useAccessToken` hook

Reply:
done